conformance/lang/expressions/transactional-expr/transactional_expr.balt

```diff
@@ -0,0 +1,571 @@
+        }
+    }
+
+    if !transactional {
+        a = a + 40;
+    }
+
+    io:println(a); // @output 110
+}
+
+transactional function f1() returns int? {
+    int? a = 10;
+    if a is () || transactional {
+        a = a + 20;
+        if a is int && transactional {
+            a = a + 30;
+        }
+    }
+
+    return a;
+}
+
+Test-Case: output
+Description: Test transactional expression as expression in if statement.
+Labels: check, commit-action, compound-assignment-stmt, error, if-else-stmt, int, logical-expr, optional-type,
+        transaction-stmt, transactional-expr
+
+function init() returns error? {
+    int a = 10;
+
+    transaction {
+        if transactional {
+            if transactional {
+                if transactional {
+                    if transactional {
+                        transaction {
+                            if transactional {
+                                a += 5;
+                            }
+                            check commit;
+                        }
+                        if transactional {
+                            if transactional {
+                                transaction {
+                                    if transactional {
+                                        a += 15;
+                                    }
+                                    check commit;
+                                }
+                                if transactional {
+                                    if transactional {
+                                        a += 20;
+                                        check commit;
+                                        if !transactional {
+                                            a += 30;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if !transactional {
+        a += 50;
+    }
+
+    io:println(a); // @output 130
+}
+
+Test-Case: output
+Description: Test transactional expression as expression in type test expression.
+Fail-Issue: ballerina-platform/ballerina-lang#33087
+Labels: boolean, int, is-expr, module-const-decl, module-type-defn, transactional-expr, union-type
+
+const TRUE = true;
+const FALSE = false;
+
+type Type1 boolean;
+
+type Type2 true|false;
+
+function init() {
+    io:println(transactional is boolean); // @output true
+    io:println(transactional is boolean|int); // @output true
+    io:println(transactional is Type1); // @output true
+    io:println(transactional is Type2); // @output true
+    io:println(transactional is TRUE|FALSE); // @output true
+}
+
+Test-Case: output
+Description: Test assigning the result of transactional expression to a variable.
+Fail-Issue: ballerina-platform/ballerina-lang#33087
+Labels: boolean, transactional-expr
+
+function init() {
+    boolean a = transactional;
```

Comment:
Enclose the assignment in a transaction as a workaround. Test in 410 would track the original issue

Reply:
Refactor where applicable